ldap/client/api/src/main/java/org/apache/directory/ldap/client/api/LdapNetworkConnection.java

```diff
@@ -599,102 +604,100 @@ public boolean connect() throws LdapException
+                continue;
+            }
 
-                    ConnectFuture futureResult = null;
-                    try
-                    {
-                        futureResult = cf.getConnectFuture().get();
-                    }
-                    catch ( InterruptedException e )
-                    {
-                        LOG.info( "InterruptedException occurred when fetching future results - Error: " + e.getMessage() );
-                        cf.getConnector().dispose();
-                        continue;
-                    }
-                    catch ( ExecutionException e )
-                    {
-                        LOG.info( "ExecutionException occurred when fetching future results - Error: " + e.getMessage() );
-                        cf.getConnector().dispose();
-                        continue;
-                    }
+            if ( result.getConnectFuture() == null || !result.getConnectFuture().isConnected() )
+            {
+                result.getConnector().dispose();
+                continue;
+            }
 
-                    if ( futureResult == null || !futureResult.isConnected() )
-                    {
-                        cf.getConnector().dispose();
-                        continue;
-                    }
+            connectionFuture = result.getConnectFuture();
+            connector = result.getConnector();
 
-                    connFuture = cf;
-                    connectionFuture = futureResult;
-                    setConnector( cf.getConnector() );
+            break;
+        }
 
-                    break breakaway;
-                }
+        for ( Future<LdapNetworkConnectionFuture> future : futures )
+        {
+            LdapNetworkConnectionFuture futureToCancel;
+            try
+            {
+                futureToCancel = future.get();
             }
-
-            for ( LdapNetworkConnectionFuture cf : connFutures )
+            catch ( InterruptedException e )
             {
-                try
-                {
-                    if ( cf.getConnectFuture().isDone() && cf.getConnectFuture().get().isConnected() )
-                    {
-                        cf.getConnectFuture().get().getSession().closeNow();
-                    }
-                }
-                catch ( InterruptedException e )
-                {
-                    LOG.debug( "InterruptedException thrown while trying to close an open session: {}", e.getMessage() );
-                }
-                catch ( ExecutionException e )
-                {
-                    LOG.debug( "ExecutionException thrown while trying to close an open session: {}", e.getMessage() );
-                }
+                LOG.info( "InterruptedException thrown while cleaning up leftover futures/connections" );
+                future.cancel( true );
+                continue;
+            }
+            catch ( ExecutionException e )
```

Comment:
Unless this is very old java version, you can catch two exceptions in the same block
`InterruptedException | ExecutionException e`